src/main/java/com/ucm/ms/accounts/controllers/CreditCardController.java

```diff
@@ -37,4 +67,119 @@ public CreditCardController(CreditCardSearch creditCardSearch) {
 			return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
 		}
 	}
-}
+	
+	
+		
+		
+		@GetMapping(path = "get_credit_cards")
+		public ResponseEntity<Collection<AccountType>> getCreditCards() {
+
+			try {
+
+				Collection<AccountType> creditCards = accountTypeDao.getCreditCards();
+				System.out.println("INSIDE ACCOUNTS CARDS");
+
+				return new ResponseEntity<>(creditCards, HttpStatus.OK);
+
+			} catch (Exception e) {
+				return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+			}
+
+		}
+		
+		
+		@PostMapping(path = "/user_credit_card_signup")
+		public void usercreditCardSignUp(@RequestParam int userId, @RequestParam String cardName) {
+
+
+			// put a check here for unique account number
+		
+			
+
+
+			Random rnd = new Random();
+			int accountNumber = rnd.nextInt(999999999);
+			
+			
+			UserAccount userAccount = new UserAccount();
+			User user = userRepository.getUserById(userId);
+
+			
+			userAccount.setAccountNumber(String.valueOf(accountNumber));
+			userAccount.setUser(user);
+			userAccount.setAccountType(accountTypeDao.getIdByName(cardName));
+			userAccount.setBalance(BigDecimal.valueOf(10000.00));
+			userAccount.setActive(true);
+
+			System.out.println("HEREEE: " + userAccount.getAccountNumber());
+
+			// ALSO ADD DATA TO CREDIT_CARD TABLE HEREE AS WELL
+			
+			CreditCard creditCard = new CreditCard();
+			AccountType accountType = accountTypeDao.getAprByName(cardName);
+			
+			Random cardNumberRand = new Random();
+			String creditCardNumber = String.valueOf(cardNumberRand.nextInt(999999999));
+
+			Random cvvRand = new Random();
+			int cvvNumber = cvvRand.nextInt(999);
+			
+			SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+			String dateFormatted = dateFormat.format(new Date());
+			
+			char charAt = dateFormatted.charAt(3);
+			
+			int charToInt = Character.getNumericValue(charAt) + 2; 
+			char intToChar=(char)(charToInt + '0');   
+			
+			
+			StringBuilder myName = new StringBuilder(dateFormatted);
+			myName.setCharAt(3, intToChar);
+			
+			
+		 
+			
+			String creditCardExpiryDateCalulation = myName.toString();
+			//Date creditCardExpiryDate;
+			LocalDate creditCardExpiryDate;
+			
+					
+					
+			//	creditCardExpiryDate = new SimpleDateFormat("yyyy-MM-dd").parse(creditCardExpiryDateCalulation);
+				creditCardExpiryDate = LocalDate.parse(creditCardExpiryDateCalulation);
+
+				creditCard.setCardNumber(creditCardNumber);
+				creditCard.setUserAccount(userAccount);
+				creditCard.setInterestRate(accountType.getApr());
+				creditCard.setCvv(cvvNumber);
+				creditCard.setExpirationDate(creditCardExpiryDate);
+				
+				
+				userAccountDao.save(userAccount);
+				creditCardDao.save(creditCard);	
+				
+		}
+		
+		
+		@GetMapping("/user_account")
```

Comment:
Why is this here? We have a userAccountsController. You can make adjustments in that file to take in the userId as a parameter there if you want.